projects/packages/forms/src/modules/form-progress-indicator/view.js

```diff
@@ -0,0 +1,23 @@
+import { getContext, store } from '@wordpress/interactivity';
+
+store( 'jetpack/form', {
+	state: {
+		get getStepProgress() {
+			const context = getContext();
+			return ( Math.max( 1, context.currentStep ) / context.maxSteps ) * 100 + '%';
+		},
+	},
+	actions: {
+		initializeProgress() {
+			// Initialize progress indicator when the form loads
+			const context = getContext();
+			// Ensure we have a valid transition value
+			if (
+				! context.transition ||
+				! [ 'none', 'fade', 'slide', 'fade-slide' ].includes( context.transition )
```

Comment:
We could start simply with `none` and `fade` for v1 to avoid spending time designing and fine-tuning them; we can then follow up with adding more of these right next up.